src/middleware.js

```diff
@@ -38,6 +40,45 @@ const expressMiddleware = ({
   }
 }
 
+/**
+ * Generates a request tracer plugin for Fastify.
+ *
+ * @param {Object} options possible options
+ * @param {boolean} options.useHeader respect request header flag
+ *                                    (default: `false`)
+ * @param {string} options.headerName request header name, used if `useHeader` is set to `true`
+ *                                    (default: `X-Request-Id`)
+ * @param {boolean} options.useFastifyRequestId respect Fastify request id flag
+ *                                    (default: `false`)
+ */
+const fastifyPlugin = ({
+  useHeader = false,
+  headerName = 'X-Request-Id',
+  useFastifyRequestId = false
+} = {}) => {
+  const plugin = (fastify, _, next) => {
+    fastify.addHook('onRequest', (request, reply, done) => {
+      let requestId
+      if (useHeader) {
+        requestId = request.headers[headerName.toLowerCase()]
+      }
+      if (useFastifyRequestId) {
+        requestId = requestId || request.id
+      }
+      requestId = requestId || uuidv1()
+
+      als.run(requestId, () => {
+        wrapHttpEmitters(request.raw, reply.res)
```

Comment:
got around to testing this, and it should be `reply.raw` as well.


```
(node:50777) [FSTDEP002] FastifyDeprecation: You are accessing the Node.js core response object via "reply.res", Use "reply.raw" instead.
    at emitWarning (/Users/simen/repos/monorepo/node_modules/fastify/lib/warnings.js:63:23)
    at _Reply.get (/Users/simen/repos/monorepo/node_modules/fastify/lib/reply.js:74:7)
    at /Users/simen/repos/monorepo/node_modules/cls-rtracer/src/rtracer.js:72:45
    at async_hooks.js:263:14
    at AsyncResource.runInAsyncScope (async_hooks.js:186:9)
    at AsyncLocalStorage.run (async_hooks.js:261:21)
    at Object.<anonymous> (/Users/simen/repos/monorepo/node_modules/cls-rtracer/src/rtracer.js:71:11)
    at hookIterator (/Users/simen/repos/monorepo/node_modules/fastify/lib/hooks.js:215:10)
    at next (/Users/simen/repos/monorepo/node_modules/fastify/lib/hooks.js:156:20)
    at hookRunner (/Users/simen/repos/monorepo/node_modules/fastify/lib/hooks.js:175:3)
```

Reply:
Which fastify version do you use? I was trying `reply.raw`, but there were problems with it.

Reply:
at least with v3, haven't tested with v2

Reply:
I've only looked at v3

Reply:
I'll try to reproduce the issue and submit a fix into Fastify. Until that let's assume that Fastify v3 is not supported (as stated in the readme).

Reply:
I've found that `reply.raw` was introduced in v3.0: https://www.fastify.io/docs/v2.14.x/Reply/

So, current code is correct for Fastify v2. I've created #35 to track v3 integration.